sqlmesh/core/engine_adapter/base.py

```diff
@@ -304,15 +305,59 @@ def replace_query(
         source_queries, columns_to_types = self._get_source_queries_and_columns_to_types(
             query_or_df, columns_to_types, target_table=table_name
         )
-        return self._create_table_from_source_queries(
-            table,
-            source_queries,
-            columns_to_types,
-            replace=True,
-            table_description=table_description,
-            column_descriptions=column_descriptions,
-            **kwargs,
+        columns_to_types = columns_to_types or self.columns(table)
+        query = source_queries[0].query_factory()
+        target_table = exp.to_table(table_name)
+        self_referencing = any(
+            quote_identifiers(table) == quote_identifiers(target_table)
```

Comment:
you're shadowing the already table var,

it should be

quote_identifiers(t) == quote_identifiers(table) for t in ...

Reply:
just stuck with targe_tablet: https://github.com/TobikoData/sqlmesh/pull/2031/commits/b2cecadd3c7c9a7d80a53df66d320ca6613ce150